src/User.ts

```diff
@@ -125,6 +102,25 @@ export class User {
     return this.wallet.decrypt(encMsg, theirPubKey)
   }
 
+  /**
+   * Encrypts and serializes the given wallet.
+   * @param tlWallet Wallet to encrypt and serialize.
+   * @param password Optional password to encrypt wallet with.
+   *                 If not specified default password is used.
+   * @param progressCallback Optional encryption progress callback.
+   */
+  public async encryptWallet(
+    tlWallet: TLWalletSchema,
+    password?: string | ((progress: number) => void),
+    progressCallback?: (progress: number) => void
+  ): Promise<string> {
+    return this.wallet.encryptWallet(
+      tlWallet,
+      typeof password === 'string' ? password : this.defaultPassword,
+      typeof password === 'function' ? password : progressCallback
```

Comment:
Are you sure about all this?

typeof password === 'string' => typeof '' === 'string' - do we want to use empty string as password?

the next line typeof password === 'function' did you mean progressCallback?

Reply:
> typeof password === 'string' => typeof '' === 'string' - do we want to use empty string as password?

It is possible with ethers js, so I would also allow it.

> the next line typeof password === 'function' did you mean progressCallback?

Wanted to give user the option to specify a callback without a password like
```js
user.encryptWallet(tlWallet, progressCallback => {...})
```
But not sure about this...